Product.cs

```diff
@@ -4,48 +4,55 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using DiscountSolution;
+using CostSolution;
+using PriceCalculatorSolution;
 
-namespace PriceCalculatorSolution
+namespace ProductSolution
 {
     public class Product
     {
         public string Name { get; set; }
         public int UPC { get; set; }
-        public static decimal BasicPrice { get; set; }
+        public decimal BasicPrice { get; set; }
 
         decimal tax;
         decimal universalDiscount;
         decimal upcDiscount;
+        decimal packagingCost = AdditionalCosts.PackagingAmount;
+        decimal transportCost = AdditionalCosts.TransportAmount;
         bool taxPriority;
 
         public Product(string name, int upc, decimal price)
         {
             Name = name;
             UPC = upc;
             BasicPrice = price;
-        }
-        public void DisplayProductPrice()
-        {
-            Console.WriteLine($"Price = {TotalPrice()}");
-            Console.WriteLine($"Total discount amount: {universalDiscount + upcDiscount}");
-        }
-        public decimal TotalPrice()
-        {
             taxPriority = PriceCalculations.TaxPriority;
-            if(taxPriority)
+            upcDiscount = Discounts.UPCProductDiscount(BasicPrice, UPC);
+            if (taxPriority)
             {
-                upcDiscount = PriceCalculations.UPCProductDiscount(BasicPrice, UPC);
-                tax = PriceCalculations.ProductTax(BasicPrice);
-                universalDiscount = PriceCalculations.UniversalProductDiscount(BasicPrice);
-                return BasicPrice + tax - universalDiscount - upcDiscount;    
+                tax = AdditionalCosts.ProductTax(BasicPrice);
+                universalDiscount = Discounts.UniversalProductDiscount(BasicPrice);
             }
             else
             {
-                upcDiscount = PriceCalculations.UPCProductDiscount(BasicPrice, UPC);
-                tax = PriceCalculations.ProductTax(BasicPrice - upcDiscount);
-                universalDiscount = PriceCalculations.UniversalProductDiscount(BasicPrice - upcDiscount);
-                return BasicPrice + tax - universalDiscount - upcDiscount;
+                tax = AdditionalCosts.ProductTax(BasicPrice - upcDiscount);
+                universalDiscount = Discounts.UniversalProductDiscount(BasicPrice - upcDiscount);
             }
         }
+        public void DisplayProductPrice()
+        {
+            Console.WriteLine($"Cost = {BasicPrice}");
+            Console.WriteLine($"Tax = {tax}");
+            Console.WriteLine($"Discounts: {universalDiscount + upcDiscount}");
+            Console.WriteLine($"Packaging: {packagingCost}");
+            Console.WriteLine($"Transport: {transportCost}");
```

Comment:
how do you handle when we don't have a tax, a discount, or costs? and the cost as we mentioned is dynamic